sidecar/sidecar.go

```diff
@@ -220,5 +232,126 @@ func startProxy(conf *config.Config) error {
 		}
 	}()
 
+	debugger := api.NewDebugAPI(nginxProxy)
+
+	a := rest.NewApi()
+	a.Use(
+		&rest.TimerMiddleware{},
+		&rest.RecorderMiddleware{},
+		&rest.RecoverMiddleware{
+			EnableResponseStackTrace: false,
+		},
+		&rest.ContentTypeCheckerMiddleware{},
+		//&middleware.RequestIDMiddleware{},
+		//&middleware.LoggingMiddleware{},
+		//middleware.NewRequireHTTPS(middleware.CheckRequest{
+		//	IsSecure: middleware.IsUsingSecureConnection,
+		//	Disabled: !conf.RequireHTTPS,
+		//}),
+	)
+
+	routes := debugger.Routes()
+	router, err := rest.MakeRouter(
+		routes...,
+	)
+	if err != nil {
+		logrus.WithError(err).Error("Could not start API server")
+		return err
+	}
+	a.SetApp(router)
+
+	go func() {
+		http.ListenAndServe(fmt.Sprintf(":%v", 6116), a.MakeHandler())
+	}()
+
 	return nil
 }
+
+// Instance TODO
+type Instance struct {
+	Tags string `json:"tags"`
+	Type string `json:"type"`
+	Host string `json:"host"`
+	Port int    `json:"port"`
+}
+
+// NGINXState nginx cached lua state
+type NGINXState struct {
+	Routes    map[string][]rules.Rule `json:"routes"`
+	Instances map[string][]Instance   `json:"instances"`
+	Actions   map[string][]rules.Rule `json:"actions"`
+}
+
+func cliCommand(command string) {
+
+	switch command {
+	case "show-state":
+		httpClient := http.Client{
+			Timeout: time.Second * 10,
+		}
+		req, err := http.NewRequest("GET", "http://localhost:6116/state", nil)
+		if err != nil {
+			fmt.Println("Error occurred building the request:", err.Error())
+			return
+		}
+
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			fmt.Println("Error occurred sending the request:", err.Error())
+			return
+		}
+
+		respBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println("Error occurred reading response body:", err.Error())
+			return
+		}
+
+		sidecarstate := struct {
+			Instances []registryclient.ServiceInstance `json:"instances"`
+			Rules     []rules.Rule                     `json:"rules"`
+		}{}
+
+		err = json.Unmarshal(respBytes, &sidecarstate)
+		if err != nil {
+			fmt.Println("Error occurred loading JSON response:", err.Error())
+			return
+		}
+
+		req, err = http.NewRequest("GET", "http://localhost:5813/a8-admin", nil)
```

Comment:
same as above. May be in the YAML file under the debug_cli, we could mention these two urls (and as part of cli flags, we can mention these two urls)